monkey/agent_plugins/credentials_collectors/chrome/src/linux_credentials_database_selector.py

```diff
@@ -1,11 +1,95 @@
+    ("Chromium", ".config/chromium"),
+]
 
-class LinuxCredentialsDatabaseSelector:
-    def __init__(self):
-        pass
 
+class LinuxCredentialsDatabaseSelector:
     def __call__(self) -> Collection[BrowserCredentialsDatabasePath]:
-        return []
+        database_paths: Set[BrowserCredentialsDatabasePath] = set()
+        for browser_database_path in self._get_browser_database_paths():
+            login_data_paths = self._get_login_data_paths(browser_database_path)
+            database_paths.update(login_data_paths)
+
+        return database_paths
+
+    def _get_browser_database_paths(self) -> Sequence[Path]:
+        database_paths = []
+
+        for username, home_dir_path in self._get_home_directories().items():
+            for browser_name, browser_path in CHROMIUM_BASED_DB_PATHS:
+                browser_db_path = home_dir_path / browser_path
+                try:
+                    if browser_db_path.exists():
+                        database_paths.append(browser_db_path)
+                except Exception as err:
+                    logger.exception(
+                        f"Failed to get {browser_name} database path for {username}: {err}"
+                    )
+
+        return database_paths
+
+    @staticmethod
+    def _get_home_directories() -> UserDirectories:
+        """
+        Retrieve all users' home directories
+        """
+        try:
+            home_dirs = {
+                p.pw_name: Path(p.pw_dir) for p in pwd.getpwall()  # type: ignore[attr-defined]
+            }
+            if "HOME" in os.environ:
+                home_path = Path(os.environ["HOME"])
+                if home_path not in home_dirs.values():
+                    home_dirs[os.getlogin()] = home_path
+
+            return home_dirs
+        except Exception:
+            logger.exception("Failed to get user directories")
+            return {}
+
+    def _get_login_data_paths(
+        self, browser_database_path: Path
+    ) -> Collection[BrowserCredentialsDatabasePath]:
+        login_data_paths: Set[BrowserCredentialsDatabasePath] = set()
+        try:
+            if (browser_database_path / LOGIN_DATABASE_FILENAME).exists():
+                login_data_paths.add(
+                    BrowserCredentialsDatabasePath(
+                        database_file_path=(browser_database_path / LOGIN_DATABASE_FILENAME),
+                        master_key=DEFAULT_MASTER_KEY,
+                    )
+                )
+
+            sub_browser_database_paths = browser_database_path.iterdir()
+            for subdir in sub_browser_database_paths:
+                login_database_path = browser_database_path / subdir / LOGIN_DATABASE_FILENAME
+                if login_database_path.exists():
+                    login_data_paths.add(
+                        BrowserCredentialsDatabasePath(
+                            database_file_path=login_database_path, master_key=DEFAULT_MASTER_KEY
+                        )
+                    )
```

Comment:
I think all of these conditions and all of the searching can just be replaced with a glob.

```python
for database_file in browser_database_path.glob('**/LOGIN_DATABASE_FILENAME'):
    login_data_paths.add(...)
```

^ Not tested.